### PR TITLE
convert radbreakout to C++

### PR DESCRIPTION
**Pull request by zingale**
this also extends the C++ model parser to handle aux and vel data

## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
depends on #1449

**Comment by zingale** [comment]
results look good: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-12-07-006/rad-breakout.html

**Comment by zingale** [comment]
note: this removes some unused initial models.  Since we don't know how these were generated, I didn't think we should keep them around.

**Comment by zingale** [comment]
previous tests: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-12-07-006/index.html
RadSuOlson compilation is now fixed:
http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-12-07-007/index.html

File: Source/driver/Castro.cpp [hunk 1]
```diff
@@ -1407,12 +1407,19 @@ Castro::initData ()
           auto r = Rad_new[mfi].array();
           auto geomdata = geom.data();
 
+          GpuArray<Real, NGROUPS+1> xnu = {0.0};
+#if NGROUPS > 1
+          ca_get_xnu(xnu.begin());
```
Comment (maxpkatz) [inline Source/driver/Castro.cpp:1412]: Let's avoid the proliferation of this Fortran API please. `xnu` is already available as an array in C++, it's just a class member so we can't pass it directly to the ParallelFor but instead need to make a local copy of it. https://github.com/AMReX-Astro/Castro/blob/main/Source/radiation/MGFLD.cpp#L326-L329

Comment (zingale) [reply Source/driver/Castro.cpp]: okay, I switched this, by making the radiation group info public, and then making a copy before passing into the rad init

**Review by maxpkatz — APPROVED** [review]

**Pull request merged**